tasks/rock-paper-scissors/src/main.rs

```diff
@@ -1,100 +1,90 @@
 extern crate rand;
+#[macro_use]
+extern crate rand_derive;
 
 use std::io;
-use rand::{Rng, thread_rng};
+use rand::Rng;
 use Choice::*;
 
-#[derive(PartialEq, Clone, Copy)]
+#[derive(PartialEq, Clone, Copy, Rand, Debug)]
 pub enum Choice {
     Rock,
     Paper,
     Scissors,
 }
 
-impl Choice {
-    fn print(&self) -> &str {
-        match *self {
-            Rock => "Rock",
-            Paper => "Paper",
-            Scissors => "Scissors",
-        }
-    }
-
-    fn rand(&mut self) {
-        match thread_rng().gen_range(0, 2) {
```

Comment:
Oh yikes I just realized this would never return 2.  I suppose the code was never tested